clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -765,6 +766,16 @@ public void shouldThrowOnFlushProducerIfProducerIsClosed() {
             fail("Should have thrown as producer is already closed");
         } catch (IllegalStateException e) { }
     }
+    
+    @Test
+    @SuppressWarnings("unchecked")
```

Comment:
We should avoid to suppress warning. Seems, you need to add generics to `ProducerRecord` in L774 to fix it?

Reply:
Thanks a lot for your review @mjsax 
If I add generics, bounded type parameters will make compilation fail, as types are checked and incompatible.
In fact, the purpose of this enhancement is that MockProducer throws the exception when types are not restricted, wdyt?

Reply:
Oh, I see. We actually want to pass in an incorrect key-type to get the ClassCastException. So you are right, for this case we cannot specify the generics. Keeping the code as-is and to suppress the warning is fine than. Totally missed this. Thanks for clarifying.